haskell/src/Stablecoin/Client/Impl.hs

```diff
@@ -0,0 +1,322 @@
+-- SPDX-FileCopyrightText: 2020 tqtezos
+-- SPDX-License-Identifier: MIT
+
+module Stablecoin.Client.Impl
+  ( AddressAndAlias(..)
+  , deploy
+
+  -- * Entrypoints
+  , pause
+  , unpause
+  , configureMinter
+  , removeMinter
+  , mint
+  , burn
+  , transferOwnership
+  , acceptOwnership
+  , changeMasterMinter
+  , changePauser
+  , setTransferlist
+
+  -- * Query commands
+  , getBalance
+  , isPaused
+  , getContractOwner
+  , getPendingContractOwner
+  , getMasterMinter
+  , getPauser
+  , getTransferlist
+  , getTotalSupply
+  ) where
+
+import Fmt (Buildable(build), pretty, (+|), (|+))
+import Lorentz (EntryPointRef(Call), HasEntryPointArg, arg, useHasEntryPointArg)
+import Michelson.Typed (Dict(..), IsoValue, fromVal, toVal)
+import qualified Michelson.Typed as T
+import Morley.Client
+  (AddressOrAlias(..), Alias, MorleyClientM, TezosClientError(UnknownAddress), getAlias,
+  getContractScript, lTransfer, originateUntypedContract, resolveAddress)
+import qualified Morley.Client as Client
+import Morley.Client.RPC (OriginationScript(OriginationScript))
+import Morley.Micheline (Expression, FromExpressionError, fromExpression)
+import Tezos.Address (Address)
+import Tezos.Core (Mutez, unsafeMkMutez)
+import Util.Named ((:!), (.!))
+
+import Lorentz.Contracts.Stablecoin
+  (pattern MasterMinterRole, pattern OwnerRole, Parameter, pattern PauserRole,
+  pattern PendingOwnerRole)
+import Stablecoin.Client.Contract
+  (InitialStorageData(..), mkInitialStorage, parseStablecoinContract)
+
+-- | An address and an optional alias, if one is found.
+data AddressAndAlias = AddressAndAlias Address (Maybe Alias)
+
+-- | Deploy the stablecoin contract.
+-- Returns the operation hash and the contract's new address.
+--
+-- Saves the contract with the given alias.
+-- If the given alias already exists, nothing happens.
+deploy :: "sender" :! AddressOrAlias -> Text -> InitialStorageData AddressOrAlias -> MorleyClientM (Text, Address)
+deploy (arg #sender -> sender) alias initialStorageData = do
+  senderAddr <- resolveAddress sender
+  stablecoinContract <- parseStablecoinContract
+  masterMinter <- resolveAddress (isdMasterMinter initialStorageData)
+  contractOwner <- resolveAddress (isdContractOwner initialStorageData)
+  pauser <- resolveAddress (isdPauser initialStorageData)
+  transferlist <- resolveAddress (isdTransferlist initialStorageData)
+
+  let initialStorageData' = initialStorageData
+        { isdMasterMinter = masterMinter
+        , isdContractOwner = contractOwner
+        , isdPauser = pauser
+        , isdTransferlist = transferlist
+        }
+
+  let initialStorage = mkInitialStorage initialStorageData'
+
+  originateUntypedContract
+    False
+    alias
+    (AddressResolved senderAddr)
```

Comment:
It seems you can pass `sender` here and avoid doing `resolveAddress sender`. It's the same logic, but a bit shorter.

Reply:
Done